blog/2023-12-18-config-entry-minor-version.md

```diff
@@ -0,0 +1,18 @@
+---
+author: Erik Montnémery
+authorURL: https://github.com/emontnemery
+title: "Config entry now supports minor versions"
+---
+
+Config entry now supports minor versions.
+
+If minor versions differ but major versions are same, integration setup will be allowed to continue even if the integration does not implement `async_migrate_entry`. This means a minor version bump is backwards compatible unlike a major version bump which causes the integration to fail setup if the user downgrades HA Core without restoring configuration from backup.
+
+### Background
+
+We have been very conservative with versioning config entry data because it breaks downgrading to an older version of Home Assistant Core. This means in most cases we don't version, and the integrations instead do a kind of soft upgrade where they may for example do `dict.get` on config entry data which was not in an initial version, transform the data during setup etc.
+
+By introducting minor versions, similar to that already offered by the storage helper, this pattern is no longer recommended.
+A bump of the minor version should be done instead whenever the newly added, or otherwise changed, data does not break older versions.
+
+More details can be found in the [core PR #105749](https://github.com/home-assistant/core/pull/105479).
```

Comment:
Maybe link to the docs section instead?

Reply:
I think we can link to both

Reply:
Did you see my comment in the merged PR about the other migration docs we should probably update too?

Reply:
https://developers.home-assistant.io/docs/config_entries_index/#migrating-config-entries-to-a-new-version is more or less empty and just links to https://developers.home-assistant.io/docs/config_entries_config_flow_handler#config-entry-migration

We should maybe move the migration documentation from config_entries_config_flow_handler to config_entries_index, it doesn't make sense to document it twice?

Reply:
Yes, I was thinking the same.